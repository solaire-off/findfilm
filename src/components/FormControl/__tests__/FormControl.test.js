import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { FormControl } from "..";

it("Current render (input) with custom style and additional classes", () => {
  const { asFragment } = render(
    <FormControl
      type="text"
      placeholder="Example placeholder"
      additionalClass="example-additional-class"
      formControlStyle="form-control--light"
      value="Example value"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it;

it("Current render (input) with default style and size", () => {
  const { asFragment } = render(
    <FormControl
      type="text"
      placeholder="Example placeholder"
      value="Example value"
      callback={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Current render (textarea) with default style and size", () => {
  const { asFragment } = render(
    <FormControl
      type="textarea"
      placeholder="Example placeholder"
      value="Example value"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Callback onChange work (textarea)", () => {
  const mockCallback = jest.fn();
  const captionToType = "Hello";
  render(
    <FormControl
      type="textarea"
      placeholder="Example placeholder"
      callback={mockCallback}
    />
  );
  userEvent.type(screen.getByRole("textbox"), captionToType);
  expect(mockCallback).toBeCalledTimes(captionToType.length);
});

it("Callback onChange work (input)", () => {
  const mockCallback = jest.fn();
  const captionToType = "Hello";
  render(
    <FormControl
      type="text"
      placeholder="Example placeholder"
      callback={mockCallback}
    />
  );
  userEvent.type(screen.getByRole("textbox"), captionToType);
  expect(mockCallback).toBeCalledTimes(captionToType.length);
});

it("Current render when disabled(textarea)", () => {
  const { asFragment } = render(
    <FormControl
      type="textarea"
      placeholder="Example placeholder"
      disabled={true}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Current render when disabled(input)", () => {
  const { asFragment } = render(
    <FormControl
      type="text"
      placeholder="Example placeholder"
      disabled={true}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
