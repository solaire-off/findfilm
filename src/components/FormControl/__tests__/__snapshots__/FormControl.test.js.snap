// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render (input) with custom style and additional classes 1`] = `
<DocumentFragment>
  <input
    class="form-control form-control--light example-additional-class"
    placeholder="Example placeholder"
    type="text"
    value="Example value"
  />
</DocumentFragment>
`;

exports[`Current render (input) with default style and size 1`] = `
<DocumentFragment>
  <input
    class="form-control  "
    placeholder="Example placeholder"
    type="text"
    value="Example value"
  />
</DocumentFragment>
`;

exports[`Current render (textarea) with default style and size 1`] = `
<DocumentFragment>
  <textarea
    class="form-control   form-control--textarea"
    placeholder="Example placeholder"
  >
    Example value
  </textarea>
</DocumentFragment>
`;

exports[`Current render when disabled(input) 1`] = `
<DocumentFragment>
  <input
    class="form-control  "
    disabled=""
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`Current render when disabled(textarea) 1`] = `
<DocumentFragment>
  <textarea
    class="form-control   form-control--textarea"
    disabled=""
  />
</DocumentFragment>
`;
